misk-core/src/test/kotlin/misk/sampling/RateLimiterTest.kt

```diff
@@ -66,53 +79,69 @@ class RateLimiterTest {
   fun `consumer faster than target rate`() {
     rateLimiter.permitsPerSecond = 2L
 
-    assertThat(rateLimiter.tryAcquire(1L, 0, TimeUnit.MILLISECONDS)).isTrue()
+    assertThat(rateLimiter.getPermitsRemaining(TimeUnit.MILLISECONDS, 499)).isEqualTo(2L)
+    assertThat(rateLimiter.tryAcquire(1L, 499, TimeUnit.MILLISECONDS)).isTrue()
     assertThat(ticker.nowMs).isEqualTo(0L)
 
-    assertThat(rateLimiter.tryAcquire(1L, 0, TimeUnit.MILLISECONDS)).isTrue()
+    assertThat(rateLimiter.getPermitsRemaining(TimeUnit.MILLISECONDS, 499)).isEqualTo(1L)
+    assertThat(rateLimiter.tryAcquire(1L, 499, TimeUnit.MILLISECONDS)).isTrue()
     assertThat(ticker.nowMs).isEqualTo(0L)
 
-    assertThat(rateLimiter.tryAcquire(1L, 0, TimeUnit.MILLISECONDS)).isFalse()
+    assertThat(rateLimiter.getPermitsRemaining(TimeUnit.MILLISECONDS, 499)).isEqualTo(0L)
+    assertThat(rateLimiter.tryAcquire(1L, 499, TimeUnit.MILLISECONDS)).isFalse()
+    assertThat(ticker.nowMs).isEqualTo(0L)
+
+    assertThat(rateLimiter.getPermitsRemaining(TimeUnit.MILLISECONDS, 499)).isEqualTo(0L)
+    assertThat(rateLimiter.tryAcquire(1L, 499, TimeUnit.MILLISECONDS)).isFalse()
     assertThat(ticker.nowMs).isEqualTo(0L)
   }
 
   @Test
   fun `rate limit increases`() {
     rateLimiter.permitsPerSecond = 2L
 
+    assertThat(rateLimiter.getPermitsRemaining(TimeUnit.MILLISECONDS, 100)).isEqualTo(2L)
     assertThat(rateLimiter.tryAcquire(1L, 100, TimeUnit.MILLISECONDS)).isTrue()
     assertThat(ticker.nowMs).isEqualTo(0L)
 
+    assertThat(rateLimiter.getPermitsRemaining(TimeUnit.MILLISECONDS, 100)).isEqualTo(1L)
     assertThat(rateLimiter.tryAcquire(1L, 100, TimeUnit.MILLISECONDS)).isTrue()
     assertThat(ticker.nowMs).isEqualTo(0L)
 
     // Exhausted.
+    assertThat(rateLimiter.getPermitsRemaining(TimeUnit.MILLISECONDS, 100)).isEqualTo(0L)
     assertThat(rateLimiter.tryAcquire(1L, 100, TimeUnit.MILLISECONDS)).isFalse()
     assertThat(ticker.nowMs).isEqualTo(0L)
 
     rateLimiter.permitsPerSecond = 10L
 
+    assertThat(rateLimiter.getPermitsRemaining(TimeUnit.MILLISECONDS, 100)).isEqualTo(1L)
```

Comment:
This assertion is currently failing because we have reached the maxNs in our current bucket. However, since our rate limiting changed, we should be allowing extension into the next bucket of time. I know we decided to not factor in permitCount, but I think this is the case where it's needed.

Reply:
The `permitCount` parameter from `tryAcquire` represents how many permits the caller is asking for. This should not have any impact on the calculation of how many permits are available.

(disussed in another forum; we're changing the calculation to be simpler, but still not using `permitCount`)